src/cpp/src/continuous_batching/pipeline_base.cpp

```diff
@@ -164,52 +165,65 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
              const std::vector<std::vector<ov::Tensor>>& images_vector,
              const std::vector<std::vector<ov::Tensor>>& videos_vector,
              const std::vector<GenerationConfig>& sampling_params,
-             const StreamerVariant& streamer)  {
+             const StreamerVariant& streamer) {
     auto generate_start_time = std::chrono::steady_clock::now();
     OPENVINO_ASSERT(m_model_input_type == ModelInputType::EMBEDDINGS);
 
     OPENVINO_ASSERT(prompts.size() == sampling_params.size(), "Number of prompts should be equal to the number of generation configs.");
-    OPENVINO_ASSERT(prompts.size() == images_vector.size() && prompts.size() == videos_vector.size(), "Number of prompts should be equal to the number of images or video vectors.");
+    if (images_vector.size() > 0)
+        OPENVINO_ASSERT(prompts.size() == images_vector.size(), "Number of prompts should be equal to the number of images vectors.");
+    if (videos_vector.size() > 0)
+        OPENVINO_ASSERT(prompts.size() == videos_vector.size(), "Number of prompts should be equal to the number of videos vectors.");
```

Comment:
To me it looks quite strange that in every function we have `images_vector` and `videos_vector`: in signatures, in body implementation. They can't co-exist right when user can pass both entities, right?
Implementation looks with duplicate functionality. Can you unify it and have just `visual_vector`, for example, and based on it characteristics you can deduce if it is an image or video.

Reply:
No, They can co-exist.
In most of real cases, customers only input images, or only input videos, I split to check them, just allow customers to pass empty vector images when they only process multiple videos.

Anyway, If you still think it is confusing or strange, I can revert them, @rkazants

Reply:
About unify them to `visual_vector`, we need to overturn the existing implementation, maybe arch needs to make a decision

Reply:
It seems to be confused  for someone, haha, just revert this changes.